.github/workflows/ci-tests.yaml

```diff
@@ -89,6 +89,22 @@ jobs:
             build-matrix: ${{ steps.build-matrix.outputs.build-matrix }}
             test-dirs-matrix: ${{ steps.dir-matrix.outputs.test-dirs-matrix }}
 
+    clang-fast-compilation:
+        # gem5 binaries built in `quick-gem5-builds` always use GCC.
+        # Clang is more strict than GCC. This job checks that gem5 compiles
+        # with Clang. It compiles build/ALL/gem5.fast to maximize the change
+        # for compilation error to be exposed.
+        runs-on: [self-hosted, linux, x64]
+        if: github.event.pull_request.draft == false
+        container: ghcr.io/gem5/clang-version-11:latest
```

Comment:
Why not do clang 16 (or the highest we support)? That seems to be where we get the most errors.

Reply:
Just wanted to make sure 16 was working first before doing this. It is so have updated.